### PR TITLE
COMMERCE-5398 When coupon code is added to an order by a non eligible account an error message should be shown

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		frutillaRule.scenario(
+			"Verify that the discount is validated by a discount validator"
+		).given(
+			"A not valid discount"
+		).and(
+			"A discount validator"
+		).when(
+			"I check if the discount is valid using the validator"
+		).then(
+			"The discount should result not valid"
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java:83]: `"The discount should result as not valid"`

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		frutillaRule.scenario(
+			"Verify that the discount is validated by a discount validator"
+		).given(
+			"A valid discount"
+		).and(
+			"A discount validator"
+		).when(
+			"I check if the discount is valid using the validator"
+		).then(
+			"The discount should be result valid"
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java:117]: `"The discount should result as valid`

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		frutillaRule.scenario(
+			"Verify that the discount is validated by a discount validator"
+		).given(
+			"A not valid discount"
+		).and(
+			"A discount validator"
+		).when(
+			"I check if the discount is valid using the validator"
+		).then(
+			"The discount should result not valid"
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java:151]: `"The discount should result as not valid"`

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+		frutillaRule.scenario(
+			"Verify that the discount is validated by a discount validator"
+		).given(
+			"A valid discount"
+		).and(
+			"A discount validator"
+		).when(
+			"I check if the discount is valid using the validator"
+		).then(
+			"The discount should be result valid"
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountValidatorTest.java:194]: `"The discount should result as valid"`

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

**Comment by marco-leo** [comment]
Need few changes discussed offline

**Pull request closed without merging**